pkgs/development/tools/ocaml/findlib/default.nix

```diff
@@ -36,15 +36,21 @@ stdenv.mkDerivation rec {
             export CAML_LD_LIBRARY_PATH="''${CAML_LD_LIBRARY_PATH-}''${CAML_LD_LIBRARY_PATH:+:}''$1/lib/ocaml/${ocaml.version}/site-lib/stublibs"
         fi
     }
-    createOcamlDestDir () {
+    exportOcamlDestDir () {
         export OCAMLFIND_DESTDIR="''$out/lib/ocaml/${ocaml.version}/site-lib/"
+    }
+    createOcamlDestDir () {
         if test -n "''${createFindlibDestdir-}"; then
           mkdir -p $OCAMLFIND_DESTDIR
         fi
     }
 
+    # run for every buildInput
     addEnvHooks "$targetOffset" addOCamlPath
+    # run before configurePhase, even without buildInputs, and not in nix-shell
     preConfigureHooks+=(createOcamlDestDir)
```

Comment:
reasoning is "before any upstream-provided code runs". It is likely to be too pessimistic. Although many configure scripts in the ocaml ecosystem are hand written shell scripts instead of autotools, so sometimes they do weird things...

If you prefer preInstall I'll change.

Reply:
It's nicer because it works in `nix-shell` properly. In #124504 I used the `preInstall` approach and it revealed no regressions back then.

Reply:
done